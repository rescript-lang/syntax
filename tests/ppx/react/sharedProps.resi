@@jsxConfig({version:4, mode: "classic"})

module V4C1: {
  @react.component(:sharedProps<string>)
  let make: (~x: string) => React.element
}

module V4C2: {
  @react.component(:sharedProps<'a>)
  let make: (~x: string) => React.element
}

module V4C3: {
  @react.component(:sharedProps<string, 'a>)
  let make: (~x: string) => React.element
}

module V4C4 : {
  @react.component(:sharedProps)
  let make: (~x: string) => React.element
}

@@jsxConfig({version:4, mode: "automatic"})

module V4A1: {
  @react.component(:sharedProps<string>)
  let make: (~x: string) => React.element
}

module V4A2: {
  @react.component(:sharedProps<'a>)
  let make: (~x: string) => React.element
}

module V4A3: {
  @react.component(:sharedProps<string, 'a>)
  let make: (~x: string) => React.element
}

module V4A4: {
  @react.component(:sharedProps)
  let make: (~x: string) => React.element
}