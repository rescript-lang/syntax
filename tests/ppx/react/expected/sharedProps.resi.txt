@@jsxConfig({version: 4, mode: "classic"})

module V4C1: {
  type props = sharedProps<string>

  let make: React.componentLike<props, React.element>
}

module V4C2: {
  type props<'a> = sharedProps<'a>

  let make: React.componentLike<props<_>, React.element>
}

module V4C3: {
  type props<'a> = sharedProps<string, 'a>

  let make: React.componentLike<props<_>, React.element>
}

module V4C4: {
  type props = sharedProps

  let make: React.componentLike<props, React.element>
}

@@jsxConfig({version: 4, mode: "automatic"})

module V4A1: {
  type props = sharedProps<string>

  let make: React.componentLike<props, React.element>
}

module V4A2: {
  type props<'a> = sharedProps<'a>

  let make: React.componentLike<props<_>, React.element>
}

module V4A3: {
  type props<'a> = sharedProps<string, 'a>

  let make: React.componentLike<props<_>, React.element>
}

module V4A4: {
  type props = sharedProps

  let make: React.componentLike<props, React.element>
}
